prediction/src/algorithms/identify/prediction.py

```diff
@@ -160,107 +199,124 @@ def predict_cubes(model_path, patient_id, magnification=1, ext_name=""):  # noqa
     if not os.path.exists(dst_dir):
         os.makedirs(dst_dir)
 
-    holdout_ext = ""
-    flip_ext = ""
-
-    dst_dir += "predictions" + str(int(magnification * 10)) + holdout_ext + flip_ext + "_" + ext_name + "/"
+    dst_dir = os.path.join(dst_dir, "predictions" + str(int(magnification * 10)) + "_" + ext_name)
     if not os.path.exists(dst_dir):
         os.makedirs(dst_dir)
 
     model = get_net(input_shape=(CUBE_SIZE, CUBE_SIZE, CUBE_SIZE, 1),
                     load_weight_path=model_path)
+
     patient_ids = [patient_id]
     for file_name in os.listdir(EXTRACTED_IMAGE_DIR):
```

Comment:
listdir returns items in nondeterminstic order; doesn't this make a difference here?

Reply:
In the original  Julian de Wit [implementation](https://github.com/juliandewit/kaggle_ndsb2017/blob/6de658e233201e573f1a6b2d150e0d51eaae9927/step3_predict_nodules.py#L198) `os.listdir` was used. The order was not necessary there scince each `file_name` represents a different patient. In the [adopted version](https://github.com/WGierke/concept-to-clinic/blob/2036289144cce659561539e2f29861648ba4842b/prediction/src/algorithms/identify/prediction.py#L251), `return df` returns a data frame only for the last patient, though. Since it's not the [original behaviour](https://github.com/juliandewit/kaggle_ndsb2017/blob/6de658e233201e573f1a6b2d150e0d51eaae9927/step3_predict_nodules.py#L315) followed by [this](https://github.com/juliandewit/kaggle_ndsb2017/blob/6de658e233201e573f1a6b2d150e0d51eaae9927/step3_predict_nodules.py#L100), I've corrected that to store all data frames in a dictionary which will be returned.

Reply:
Can you add this as a comment to the code itself? This (awesome!) comment will get lost here, alas … :(

Reply:
I've added [the comment](https://github.com/concept-to-clinic/concept-to-clinic/pull/172/files#diff-632542ba71c59c64b64a4bc113a8c4b6R215) but I'm not completely sure that I've done it correctly in the code base.